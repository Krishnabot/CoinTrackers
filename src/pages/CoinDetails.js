import React from 'react';
import { useLocation } from 'react-router-dom';
import '../CSS/SingleCoin.css';
import statsIcon from '../assets/image/stats2.png';

const CollectionDetails = () => {
  const location = useLocation();

  const coin = location?.state?.coin;

  return (
    <div className="coins-info">
      <div className="coin-text">
        <div className="coin-name">
          <h2>
            Coin Name =
            {' '}
            {coin.name}
          </h2>
          <span>
            ( Coin Symbol =
            {coin.symbol}
            )
          </span>
        </div>
        <div className="coin-health">
          <span className="price">
            Price =
            $
            {coin.price}
            {' '}
            USD
          </span>
          <span className="coin-rank">
            Rank =
            {coin.rank}
            <i>
              <img src={statsIcon} alt="" />
            </i>
          </span>
        </div>
      </div>

      <div className="stats-container">
        <div className="stats-title">Overall Coin Performance</div>
        <div className="stats-wrap">
          <div className="stat-item">
            <span>Change (24h)</span>
            {coin.percent_change_24h}
            {' '}
            %
          </div>
          <div className="stat-item">
            <span>Change (1h)</span>
            {coin.percent_change_1h}
            %
          </div>
          <div className="stat-item">
            <span>Change (7d)</span>
            {coin.percent_change_7d}
            %
          </div>
          <div className="stat-item">
            <span>BTC</span>
            {coin.price_btc}
          </div>
          <div className="stat-item">
            <span>Market Cap $</span>
            {coin.market_cap_usd}
          </div>
          <div className="stat-item">
            <span>Volume (24h)</span>
            {coin.volume24}
          </div>
          <div className="stat-item">
            <span>Volume (24h A)</span>
            {coin.volume24a}
          </div>
          <div className="stat-item">
            <span>Coin Supply</span>
            {coin.csupply}
          </div>
        </div>
      </div>
    </div>
  );
};

export default CollectionDetails;
