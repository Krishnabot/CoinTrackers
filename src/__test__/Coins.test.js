import React from 'react';
import { render } from '@testing-library/react';
import { BrowserRouter as Router, Route, Routes } from 'react-router-dom';
import { Provider } from 'react-redux';
import { createBrowserHistory } from 'history';
import CoinDetails from '../pages/CoinDetails';
import store from '../redux/configureStore';

describe('Test for coins', () => {
  test('whether it matches snapshots or not', () => {
    const history = createBrowserHistory();
    const coin = {
      id: '80',
      symbol: 'ETH',
      name: 'Ethereum',
      nameid: 'ethereum',
      rank: 2,
      price_usd: '1428.98',
      percent_change_24h: '-1.48',
      percent_change_1h: '0.24',
      percent_change_7d: '-17.00',
      price_btc: '0.071292',
      market_cap_usd: '174627657078.98',
      volume24: '12027472426.885479',
      volume24a: '17284898737.62316',
      csupply: '1122204013.00',
      tsupply: '122204013',
      msupply: '',
    };

    history.push('/coindetail', { coin });
    const tree = render(
      <React.StrictMode>
        <Provider store={store}>
          <Router history={history}>
            <Routes>
              <Route path="/coindetail" element={<CoinDetails />} />
            </Routes>
          </Router>
        </Provider>
      </React.StrictMode>,
    );
    expect(tree).toMatchSnapshot();
  });
});
