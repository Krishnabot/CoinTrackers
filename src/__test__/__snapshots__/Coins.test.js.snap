// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Coins test snapshot for coins 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="coins-info"
      >
        <div
          class="coin-text"
        >
          <div
            class="coin-name"
          >
            <h2>
              Coin Name =
               
              Ethereum
            </h2>
            <span>
              ( Coin Symbol =
              ETH
              )
            </span>
          </div>
          <div
            class="coin-health"
          >
            <span
              class="price"
            >
              Price = $
               
              USD
            </span>
            <span
              class="coin-rank"
            >
              Rank =
              2
              <i>
                <img
                  alt=""
                  src="stats2.png"
                />
              </i>
            </span>
          </div>
        </div>
        <div
          class="stats-container"
        >
          <div
            class="stats-title"
          >
            Overall Coin Performance
          </div>
          <div
            class="stats-wrap"
          >
            <div
              class="stat-item"
            >
              <span>
                Change (24h)
              </span>
              -1.48
               
              %
            </div>
            <div
              class="stat-item"
            >
              <span>
                Change (1h)
              </span>
              0.24
              %
            </div>
            <div
              class="stat-item"
            >
              <span>
                Change (7d)
              </span>
              -17.00
              %
            </div>
            <div
              class="stat-item"
            >
              <span>
                BTC
              </span>
              0.071292
            </div>
            <div
              class="stat-item"
            >
              <span>
                Market Cap $
              </span>
              174627657078.98
            </div>
            <div
              class="stat-item"
            >
              <span>
                Volume (24h)
              </span>
              12027472426.885479
            </div>
            <div
              class="stat-item"
            >
              <span>
                Volume (24h A)
              </span>
              17284898737.62316
            </div>
            <div
              class="stat-item"
            >
              <span>
                Coin Supply
              </span>
              1122204013.00
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="coins-info"
    >
      <div
        class="coin-text"
      >
        <div
          class="coin-name"
        >
          <h2>
            Coin Name =
             
            Ethereum
          </h2>
          <span>
            ( Coin Symbol =
            ETH
            )
          </span>
        </div>
        <div
          class="coin-health"
        >
          <span
            class="price"
          >
            Price = $
             
            USD
          </span>
          <span
            class="coin-rank"
          >
            Rank =
            2
            <i>
              <img
                alt=""
                src="stats2.png"
              />
            </i>
          </span>
        </div>
      </div>
      <div
        class="stats-container"
      >
        <div
          class="stats-title"
        >
          Overall Coin Performance
        </div>
        <div
          class="stats-wrap"
        >
          <div
            class="stat-item"
          >
            <span>
              Change (24h)
            </span>
            -1.48
             
            %
          </div>
          <div
            class="stat-item"
          >
            <span>
              Change (1h)
            </span>
            0.24
            %
          </div>
          <div
            class="stat-item"
          >
            <span>
              Change (7d)
            </span>
            -17.00
            %
          </div>
          <div
            class="stat-item"
          >
            <span>
              BTC
            </span>
            0.071292
          </div>
          <div
            class="stat-item"
          >
            <span>
              Market Cap $
            </span>
            174627657078.98
          </div>
          <div
            class="stat-item"
          >
            <span>
              Volume (24h)
            </span>
            12027472426.885479
          </div>
          <div
            class="stat-item"
          >
            <span>
              Volume (24h A)
            </span>
            17284898737.62316
          </div>
          <div
            class="stat-item"
          >
            <span>
              Coin Supply
            </span>
            1122204013.00
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Test for coins whether it matches snapshots or not 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="coins-info"
      >
        <div
          class="coin-text"
        >
          <div
            class="coin-name"
          >
            <h2>
              Coin Name =
               
              Ethereum
            </h2>
            <span>
              ( Coin Symbol =
              ETH
              )
            </span>
          </div>
          <div
            class="coin-health"
          >
            <span
              class="price"
            >
              Price = $
               
              USD
            </span>
            <span
              class="coin-rank"
            >
              Rank =
              2
              <i>
                <img
                  alt=""
                  src="stats2.png"
                />
              </i>
            </span>
          </div>
        </div>
        <div
          class="stats-container"
        >
          <div
            class="stats-title"
          >
            Overall Coin Performance
          </div>
          <div
            class="stats-wrap"
          >
            <div
              class="stat-item"
            >
              <span>
                Change (24h)
              </span>
              -1.48
               
              %
            </div>
            <div
              class="stat-item"
            >
              <span>
                Change (1h)
              </span>
              0.24
              %
            </div>
            <div
              class="stat-item"
            >
              <span>
                Change (7d)
              </span>
              -17.00
              %
            </div>
            <div
              class="stat-item"
            >
              <span>
                BTC
              </span>
              0.071292
            </div>
            <div
              class="stat-item"
            >
              <span>
                Market Cap $
              </span>
              174627657078.98
            </div>
            <div
              class="stat-item"
            >
              <span>
                Volume (24h)
              </span>
              12027472426.885479
            </div>
            <div
              class="stat-item"
            >
              <span>
                Volume (24h A)
              </span>
              17284898737.62316
            </div>
            <div
              class="stat-item"
            >
              <span>
                Coin Supply
              </span>
              1122204013.00
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="coins-info"
    >
      <div
        class="coin-text"
      >
        <div
          class="coin-name"
        >
          <h2>
            Coin Name =
             
            Ethereum
          </h2>
          <span>
            ( Coin Symbol =
            ETH
            )
          </span>
        </div>
        <div
          class="coin-health"
        >
          <span
            class="price"
          >
            Price = $
             
            USD
          </span>
          <span
            class="coin-rank"
          >
            Rank =
            2
            <i>
              <img
                alt=""
                src="stats2.png"
              />
            </i>
          </span>
        </div>
      </div>
      <div
        class="stats-container"
      >
        <div
          class="stats-title"
        >
          Overall Coin Performance
        </div>
        <div
          class="stats-wrap"
        >
          <div
            class="stat-item"
          >
            <span>
              Change (24h)
            </span>
            -1.48
             
            %
          </div>
          <div
            class="stat-item"
          >
            <span>
              Change (1h)
            </span>
            0.24
            %
          </div>
          <div
            class="stat-item"
          >
            <span>
              Change (7d)
            </span>
            -17.00
            %
          </div>
          <div
            class="stat-item"
          >
            <span>
              BTC
            </span>
            0.071292
          </div>
          <div
            class="stat-item"
          >
            <span>
              Market Cap $
            </span>
            174627657078.98
          </div>
          <div
            class="stat-item"
          >
            <span>
              Volume (24h)
            </span>
            12027472426.885479
          </div>
          <div
            class="stat-item"
          >
            <span>
              Volume (24h A)
            </span>
            17284898737.62316
          </div>
          <div
            class="stat-item"
          >
            <span>
              Coin Supply
            </span>
            1122204013.00
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
